specs/consensus.md

```diff
@@ -71,16 +85,149 @@ Consensus Rules
+
+The last [commit](./data_structures.md#commit) `block.lastCommit` (`lastCommit` for short) is processed next. This is the Tendermint commit (i.e. polka of votes) _for the previous block_. For previous block `prev` and previous block header `prev.header`, the following checks must be `true`:
+
+1. `lastCommit.height` == `prev.header.height`.
+1. `lastCommit.round` >= `1`.
+1. `lastCommit.blockID` == the [block ID](./data_structures.md#blockid) of `prev`.
+1. Length of `lastCommit.signatures` <= [`MAX_VALIDATORS`](#constants).
+1. Each of `lastCommit.signatures` must be a valid [CommitSig](./data_structures.md#commitsig)
+1. The sum of the votes for `prev` in `lastCommit` must be at least 2/3 (rounded up) of the voting power of `prev`'s next validator set.
+
+### `block.availableData`
+
+The block's [available data](./data_structures.md#availabledata) (analogous to transactions in contemporary blockchain designs) `block.availableData` (`availableData` for short) is finally processed. The [list of share rows](./networking.md#availabledata) is parsed into the [actual data structures](./data_structures.md#availabledata) using the reverse of [the process to encode available data into shares](./data_structures.md#arranging-available-data-into-shares); if parsing fails here, the block is invalid.
+
+Once parsed, the following checks must be `true`:
 
-#### Validators and Delegations
+1. The commitments of the [erasure-coded extended](./data_structures.md#2d-reed-solomon-encoding-scheme) `availableData` must match those in `header.availableDataHeader`. Implicitly, this means that both rows and columns must be ordered lexicographically by namespace ID since they are committed to in a [Namespace Merkle Tree](data_structures.md#namespace-merkle-tree).
+1. Length of `availableData.intermediateStateRootData` == length of `availableData.transactionData` + length of `availableData.evidenceData`.
 
-A transaction `tx` that requests a new validator initializes a new [Validator](data_structures.md#validator) leaf in the inactive validators subtree for that account as follows:
+## State Transitions
+
+Once the basic structure of the block [has been validated](#block-structure), state transitions must be applied to compute the new state and state root.
+
+For this section, the variable `state` represents the [state tree](./data_structures.md#state), with `state.accounts[k]`, `state.inactiveValidatorSet[k]`, and `state.activeValidatorSet[k]` being shorthand for the leaf in the state tree in the [accounts, inactive validator set, and active validator set subtrees](./data_structures.md#state) with [pre-hashed key](./data_structures.md#state) `k`. E.g. `state.accounts[a]` is shorthand for `state[(ACCOUNTS_SUBTREE_ID << 8*(32-STATE_SUBTREE_RESERVED_BYTES)) | ((-1 >> 8*STATE_SUBTREE_RESERVED_BYTES) & hash(a))]`.
+
+### `block.availableData.evidenceData`
+
+Evidence is the first set of state transitions that are applied, and represent proof of validator misbehavior.
```

Comment:
Don't you mean the first set that represent proof of validator misbehaviour? Can you expand on what this evidence is for, and why it needs to be committed in the header?

Reply:
I don't think so? I mean that they're the first set of state transitions that are applied. I.e. each evidence is a state transition, and evidence is applied before transactions. I'd prefer to relegate full explanation of what evidence is to a future issues, e.g. #23.

Reply:
I see, I thought it meant that the evidence itself is the first set of transactions that were applied, rather that it meaning that the evidence should be applied first. Is evidence really a state transition to be applied?

Reply:
Right, evidence causes slashing, which is a state transition. Wording could be improved on that. More context: https://github.com/lazyledger/lazyledger-specs/pull/89#discussion_r521342908

Reply:
Clarified wording in 37d31ca.